File: Recursion/python/factorial.py
class Factorial:
    def __init__(self):
        pass

    def calculate_factorial(self, n: int) -> int:
        print("This program is calculating Factorial({})".format(n))
        if n == 0:
            print("Factorial({}) is {}".format(n, 1))
            return 1
        f = n * self.calculate_factorial(n - 1)
        print("Factorial({}) is {}".format(n, f))
        return f

File: Recursion/cpp/Factorial.cpp
#include "Factorial.h"

Factorial::Factorial()
{

}

int Factorial::calculateFactorial(const int& n)
{
	cout << "This program is calculating Factorial(" << n << ")\n";
	if (n == 0)
	{
		cout << "Factorial(" << n << ") is " << 1 << "\n";
		return 1;
	}
	int f = n * calculateFactorial(n - 1);
	cout << "Factorial(" << n << ") is " << f << "\n";
	return f;
}
